solver/solver.go

```diff
@@ -49,7 +49,7 @@ func Solve(ctx context.Context, syncer *diff.Syncer,
 			print.CreatePrintln("creating", e.Kind, c.Console())
 		case crud.Update:
 			var diffString string
-			if reflect.TypeOf(e.OldObj).Name() == reflect.TypeOf(&state.Document{}).Name() {
+			if reflect.TypeOf(e.OldObj).ConvertibleTo(reflect.TypeOf(&state.Document{})) {
```

Comment:
```patch
-                       if reflect.TypeOf(e.OldObj).Name() == reflect.TypeOf(&state.Document{}).Name() {
+                       // TODO handle nil?
+                       oldName := reflect.Indirect(reflect.ValueOf(e.OldObj)).Type().Name()
+                       documentTypeName := reflect.TypeOf(state.Document{}).Name()
+                       if oldName == documentTypeName {

```
I think you need to compare types of values and not pointers to values.

Reply:
Let's unit test this. Please move that comparison to a func isDocument(oldobj interface{}) bool.
Documentation says that reflect.Type are comparable using ==. Let's use that for equality checking.

(copied from internal slack)